pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -429,6 +429,9 @@ exponent:
 	}
 
 exit:
+	if buffer.Len() == 0 {
+		return LexError, nil
```

Comment:
We don't seem to return `nil` on the other return paths with errors. We should make sure this won't cause some panic somewhere.